src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1203,10 +1203,21 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                             }
 
                             var factory = new SyntheticBoundNodeFactory(methodSymbol, syntax, compilationState, diagsForCurrentMethod);
-                            var boundStatementsWithNullCheck = LocalRewriter.ConstructNullCheckedStatementList(methodSymbol.Parameters, boundStatements, factory);
-                            if (!boundStatementsWithNullCheck.IsDefault)
+                            if (!methodSymbol.IsIterator)
```

Comment:
I would add a comment here about not handling iterators because iterators are handled in the IteratorRewriter